chia/_tests/pools/test_pool_cmdline.py

```diff
@@ -1,20 +1,1050 @@
+
+        wallet_id = await create_new_plotnft(wallet_environments)
+        pw_info, _ = await wallet_rpc.pw_status(wallet_id)
+
+        await ShowPlotNFTCMD(
+            rpc_info=NeedsWalletRPC(
+                context=context,
+                client_info=client_info,
+            ),
+            id=wallet_id,
+        ).run()
+        out, _err = capsys.readouterr()
+        assert "Current state: FARMING_TO_POOL" in out
+        assert f"Wallet ID: {wallet_id}" in out
+        assert f"Launcher ID: {pw_info.launcher_id.hex()}" in out
+
+
+@pytest.mark.parametrize(
+    "wallet_environments",
+    [
+        {
+            "num_environments": 1,
+            "blocks_needed": [10],
+            "trusted": True,
+            "reuse_puzhash": False,
+        }
+    ],
+    indirect=True,
+)
+@pytest.mark.anyio
+async def test_plotnft_cli_leave(
+    wallet_environments: WalletTestFramework,
+) -> None:
+    wallet_state_manager: WalletStateManager = wallet_environments.environments[0].wallet_state_manager
+    wallet_rpc: WalletRpcClient = wallet_environments.environments[0].rpc_client
+    client_info: WalletClientInfo = WalletClientInfo(
+        wallet_rpc,
+        wallet_state_manager.root_pubkey.get_fingerprint(),
+        wallet_state_manager.config,
+    )
+
+    with pytest.raises(CliRpcConnectionError, match="No pool wallet found"):
+        await LeavePlotNFTCMD(
+            rpc_info=NeedsWalletRPC(
+                client_info=client_info,
+            ),
+            id=None,
+            dont_prompt=True,
+        ).run()
+
+    with pytest.raises(CliRpcConnectionError, match="is not a pool wallet"):
+        await LeavePlotNFTCMD(
+            rpc_info=NeedsWalletRPC(
+                client_info=client_info,
+            ),
+            id=15,
+            dont_prompt=True,
+        ).run()
+
+    wallet_id = await create_new_plotnft(wallet_environments)
+
+    await LeavePlotNFTCMD(
+        rpc_info=NeedsWalletRPC(
+            client_info=client_info,
+        ),
+        id=wallet_id,
+        dont_prompt=True,
+    ).run()
+
+    await wallet_environments.process_pending_states(
+        [
+            WalletStateTransition(
+                pre_block_balance_updates={
+                    1: {
+                        "<=#spendable_balance": 1,
+                        "<=#max_send_amount": 1,
+                        "pending_coin_removal_count": 0,
+                    },
+                    2: {"pending_coin_removal_count": 1},
+                },
+                post_block_balance_updates={
+                    1: {
+                        "<=#pending_coin_removal_count": 1,
+                    },
+                    2: {"pending_coin_removal_count": -1},
+                },
+            )
+        ]
+    )
+
+    await verify_pool_state(wallet_rpc, wallet_id, PoolSingletonState.LEAVING_POOL)
+
+    await wallet_environments.full_node.farm_blocks_to_puzzlehash(count=12, guarantee_transaction_blocks=True)
```

Comment:
Is this... based on something?  If this is just a method to try not to be flaky, I think we should pick something more concrete to at least `time_out_assert`.

Reply:
Similar concerns elsewhere in the file

Reply:
when you leave it takes several blocks before you actually leave (whatever relative_lock_height is set to). I use a relative block height of 5 - and I guess I could use that more consistently with a global or something.

Reply:
You do have to farm blocks though - you can't just wait